From cd05ec78c8374980362347a226d9c0cc5f6bb624 Mon Sep 17 00:00:00 2001
From: Agustin Martin Domingo <agmartin@debian.org>
Date: Tue, 5 Apr 2016 18:43:13 +0200
Subject: fmt_latex2e.pl: Handle perl 5.22 deprecation of unescaped left brace
 in regular expressions.

This is to deal with new perl 5.22 warnings like

 Unescaped left brace in regex is deprecated, passed through in regex;
 marked by <-- HERE in m/\\nameurl{ <-- HERE (.*)}{(.*)}/
 at /usr/share/linuxdoc-tools/fmt/fmt_latex2e.pl line 287.

For consistency, right brackets are also escaped, although they
currently don't trigger any warning.

Signed-off-by: Agustin Martin Domingo <agmartin@debian.org>
---
 lib/fmt/fmt_latex2e.pl | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/fmt/fmt_latex2e.pl b/lib/fmt/fmt_latex2e.pl
index 050cf06..69d8e62 100644
--- a/lib/fmt/fmt_latex2e.pl
+++ b/lib/fmt/fmt_latex2e.pl
@@ -284,11 +284,11 @@ $latex2e->{postASP} = sub
       push @texlines, $_;
       # and check for nameurl
       if ( /\\nameurl/ ){
-	($urlid, $urlnam) = ($_ =~ /\\nameurl{(.*)}{(.*)}/);
+	($urlid, $urlnam) = ($_ =~ /\\nameurl\{(.*)\}\{(.*)\}/);
 	print $urlnum . ": " . $urlid . "\n" if ( $global->{debug} );

 	$urldef = latex2e_defnam($urlnum) . "url";
-	s/\\nameurl{.*}{.*}/{\\em $urlnam} {\\tt \\$urldef}/;
+	s/\\nameurl\{.*\}\{.*\}/{\\em $urlnam} {\\tt \\$urldef}/;
 	push @urlnames, $_;
 	push @urldefines, "\\urldef{\\$urldef} \\url{$urlid}\n";
 	$urlnum++;
@@ -367,12 +367,12 @@ $latex2e->{postASP} = sub
 	    $_ = $_ . "\\makeindex\n" if ($latex2e->{makeindex});
 	  }
 	  # Set correct DTD name
-	  elsif (/^\\usepackage{\@LINUXDOC_DTD\@-sgml}/) {
+	  elsif (/^\\usepackage\{\@LINUXDOC_DTD\@-sgml\}/) {
 	    my $dtd = $global->{"dtd"};
 	    s/\@LINUXDOC_DTD\@/$dtd/;
 	  }
 	  # Set correct babel options
-	  elsif (/^\\usepackage\[\@BABELOPTIONS\@\]{babel}/) {
+	  elsif (/^\\usepackage\[\@BABELOPTIONS\@\]\{babel\}/) {
 	    if ( $babeloptions ) {
 	      s/\@BABELOPTIONS\@/$babeloptions/;
 	    } else {
